### PR TITLE
Ccip-4506 log improvements in deployment code

### PR DESCRIPTION
**Pull request by AnieeG**

### CODE REVIEW

File: deployment/ccip/changeset/cs_deploy_chain.go [hunk 1]
```diff
@@ -286,16 +284,15 @@ func deployChainContracts(
 				}
 			})
 		if err != nil {
-			e.Logger.Errorw("Failed to deploy RMNProxyNew", "err", err)
+			e.Logger.Errorw("Failed to deploy RMNProxyNew", "chain", chain.Selector, "err", err)
```
Comment (connorwstein) [inline deployment/ccip/changeset/cs_deploy_chain.go]: name?

File: deployment/ccip/changeset/cs_deploy_chain.go [hunk 2]
```diff
@@ -451,16 +443,17 @@ func deployChainContracts(
...
+	e.Logger.Infow("Added fee quoter authorized callers", "chain", chain.Name, "callers", offRampContract.Address(), chain.DeployerKey.From)
 	tx, err = nmContract.ApplyAuthorizedCallerUpdates(chain.DeployerKey, nonce_manager.AuthorizedCallersAuthorizedCallerArgs{
 		AddedCallers: []common.Address{offRampContract.Address(), onRampContract.Address()},
 	})
 	if _, err := deployment.ConfirmIfNoError(chain, tx, err); err != nil {
-		e.Logger.Errorw("Failed to update nonce manager with ramps", "err", err)
+		e.Logger.Errorw("Failed to update nonce manager with ramps", "chain", chain.Name, "err", err)
 		return err
 	}
+	e.Logger.Infow("Added nonce manager authorized callers", "chain", chain.Name, "callers", offRampContract.Address(), onRampContract.Address())
```
Comment (connorwstein) [inline deployment/ccip/changeset/cs_deploy_chain.go]: doesn't the Infow one have to have the form `"string", "key", value, "key", value...`?

File: deployment/ccip/changeset/cs_home_chain.go [hunk 3]
```diff
@@ -152,10 +153,9 @@ func deployHomeChain(
 			}
 		})
 	if err != nil {
-		lggr.Errorw("Failed to deploy CCIPHome", "err", err)
+		lggr.Errorw("Failed to deploy CCIPHome", "chain", chain.Name, "err", err)
 		return nil, err
 	}
-	lggr.Infow("deployed CCIPHome", "addr", ccipHome.Address)
```
Comment (connorwstein) [inline deployment/ccip/changeset/cs_home_chain.go:158]: I think we want all the positive logs too

File: deployment/ccip/changeset/cs_initial_add_chain.go [hunk 4]
```diff
@@ -267,7 +269,7 @@ func addChainConfig(
 	if _, err := deployment.ConfirmIfNoError(h, tx, err); err != nil {
 		return ccip_home.CCIPHomeChainConfigArgs{}, err
 	}
-	lggr.Infow("Applied chain config updates", "chainConfig", chainConfig)
+	lggr.Infow("Applied chain config updates", "homeChain", h.Name, "addedChain", chainSelector, "chainConfig", chainConfig)
```
Comment (connorwstein) [inline deployment/ccip/changeset/cs_initial_add_chain.go]: added chain name?

File: deployment/ccip/changeset/cs_prerequisites.go [hunk 5]
```diff
@@ -245,10 +242,9 @@ func deployPrerequisiteContracts(e deployment.Environment, ab deployment.Address
 				}
 			})
 		if err != nil {
-			lggr.Errorw("Failed to deploy weth9", "err", err)
+			lggr.Errorw("Failed to deploy weth9", "chain", chain.Name, "err", err)
 			return err
 		}
-		lggr.Infow("deployed weth9", "addr", weth.Address)
```
Comment (connorwstein) [inline deployment/ccip/changeset/cs_prerequisites.go:251]: yeah I like the positive logs, let add em everywhere

File: deployment/environment.go [hunk 6]
```diff
@@ -50,6 +50,7 @@ type OffchainClient interface {
 type Chain struct {
 	// Selectors used as canonical chain identifier.
 	Selector uint64
+	Name     string
```
Comment (connorwstein) [inline deployment/environment.go]: I worry this could be populated wrong, can we leave it as is and derive from selector?

**Review by rstout — DISMISSED** [review]

File: deployment/ccip/changeset/cs_prerequisites.go (hunk 5 above)
Comment (AnieeG) [reply deployment/ccip/changeset/cs_prerequisites.go:251]: Added with deployContract - https://github.com/smartcontractkit/chainlink/blob/CCIP-4506-log-improvements/deployment/helpers.go#L149

File: deployment/environment.go (hunk 6 above)
Comment (AnieeG) [reply deployment/environment.go]: I can move the name population to deployment.environment ow for every function we will have to call the chaininfo function,

Comment (connorwstein) [reply deployment/environment.go]: could we just have a method on Chain called Name() which does the derivation?

File: deployment/environment.go [hunk 7]
```diff
@@ -50,12 +50,26 @@ type OffchainClient interface {
...
 	Selector uint64
+	name     string
 	Client   OnchainClient
 	// Note the Sign function can be abstract supporting a variety of key storage mechanisms (e.g. KMS etc).
 	DeployerKey *bind.TransactOpts
 	Confirm     func(tx *types.Transaction) (uint64, error)
 }
 
+func (c Chain) Name() string {
+	return c.name
```
Comment (connorwstein) [inline deployment/environment.go]: oh I was thinking of just inlining 
```
func (c Chain) Name() string {
	chainInfo, err := ChainInfo(c.Selector)
	if err != nil {
		// we should never get here, if the selector is invalid it should not be in the environment
		panic(err)
	}
	return chainInfo.Name
}
```
its just a map lookup inside there anyways

**Review by connorwstein — APPROVED** [review]

**Review by connorwstein — DISMISSED** [review]

**Review by mateusz-sekara — DISMISSED** [review]

File: deployment/ccip/changeset/cs_home_chain.go (hunk 3 above)
Comment (makramkd) [reply deployment/ccip/changeset/cs_home_chain.go:158]: According to @AnieeG they were moved to `DeployContract` func

**Review by makramkd — APPROVED** [review]

**Review by mateusz-sekara — APPROVED** [review]

**Pull request merged**